packages/esm-patient-orders-app/src/order-cancellation-form/cancel-order-form.component.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import {
+  Button,
+  TextArea,
+  ButtonSet,
+  Column,
+  Form,
+  InlineNotification,
+  Stack,
+  DatePicker,
+  DatePickerInput,
+  InlineLoading,
+} from '@carbon/react';
+import { showSnackbar, useLayoutType } from '@openmrs/esm-framework';
+import { type DefaultWorkspaceProps, usePatientOrders, type Order } from '@openmrs/esm-patient-common-lib';
+import styles from './cancel-order-form.scss';
+import { cancelOrder } from './cancel-order.resource';
+import dayjs from 'dayjs';
+
+interface OrderCancellationFormProps extends DefaultWorkspaceProps {
+  order: Order;
+}
+
+const OrderCancellationForm: React.FC<OrderCancellationFormProps> = ({
+  order,
+  patientUuid,
+  closeWorkspace,
+  promptBeforeClosing,
+}) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [showErrorNotification, setShowErrorNotification] = useState(false);
+  const { mutate } = usePatientOrders(patientUuid);
+
+  const cancelOrderSchema = useMemo(() => {
+    return z.object({
+      cancellationDate: z
+        .date({
+          required_error: t('requiredField', 'This field is required'),
+        })
+        .refine((date) => date >= dayjs().startOf('day').toDate(), {
+          message: t('dateCannotBeBeforeToday', 'Date cannot be before today'),
+        }),
+      reasonForCancellation: z.string({
+        required_error: t('requiredField', 'This field is required'),
+      }),
+    });
+  }, []);
+
+  type CancelOrderFormData = z.infer<typeof cancelOrderSchema>;
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, isDirty },
+  } = useForm<CancelOrderFormData>({
+    mode: 'all',
+    resolver: zodResolver(cancelOrderSchema),
+  });
+
+  function onError(err) {
+    if (err?.oneFieldRequired) {
+      setShowErrorNotification(true);
+    }
+  }
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty]);
+
+  const cancelOrderRequest = useCallback((data: CancelOrderFormData) => {
+    const formData = data;
+    setShowErrorNotification(false);
+    setIsSubmitting(true);
+
+    const payload = {
+      fulfillerStatus: 'DECLINED',
+      fulfillerComment: formData.reasonForCancellation,
+    };
+
+    const abortController = new AbortController();
```

Comment:
Use the `useAbortController()` hook to get an actual abort controller.

Reply:
Could we document this [here](https://o3-docs.openmrs.org/docs/coding-conventions#data-fetching), @ibacher?